crates/trie/parallel/benches/root.rs

```diff
@@ -37,7 +38,7 @@ pub fn calculate_state_root(c: &mut Criterion) {
             provider_rw.commit().unwrap();
         }
 
-        let view = ConsistentDbView::new(provider_factory.clone(), None);
+        let view = OverlayStateProviderFactory::new(provider_factory.clone());
```

Comment:
If we could add a blanket implementation for TrieCursorFactory and HashedCursorFactory on DBProvider, then (combined with the DatabaseProviderROFactory blanket implementation in this PR) I don't think wrapping this factory would be necessary. But doing that created some import cycles so I left it alone for now.